#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "hit"

gen = ParameterGenerator()

gen.add("radius_KP", double_t, 0, "radius_KP", 0, -100, 100) 
gen.add("radius_KI", double_t, 0, "radius_KI", 0, -100, 100)
gen.add("radius_KD", double_t, 0, "radius_KD", 0, -100, 100)

gen.add("angle_KP", double_t, 0, "angle_KP", 0, -100, 100) 
gen.add("angle_KI", double_t, 0, "angle_KI", 0, -100, 100)
gen.add("angle_KD", double_t, 0, "angle_KD", 0, -100, 100)

gen.add("max_velocity_radius", double_t, 0, "max_velocity_radius", 0, 0, 4)
gen.add("max_velocity_angle", double_t, 0, "max_velocity_angle", 0, 0, 4)

gen.add("max_radius_error", double_t, 0, "max_radius_error", 0, 0, 10)
gen.add("max_angle_error", double_t, 0, "max_angle_error", 0, 0, 90)

gen.add("PID_period", double_t, 0, "PID_period", 0, 0, 1)

exit(gen.generate(PACKAGE, "hit_node", "PID"))